sunkit_image/radial.py

```diff
@@ -695,46 +687,69 @@ def rhef(
+    # Get the radii for every pixel, ensuring units are correct (in terms of pixels or solar radii)
+    map_r = find_pixel_radii(smap)
 
+    # Automatically generate radial bin edges if none are provided
     if radial_bin_edges is None:
-        radial_bin_edges = equally_spaced_bins(0, 2, smap.data.shape[0] // 2)
-        radial_bin_edges *= u.R_sun
+        radial_bin_edges = equally_spaced_bins(0, 10 * np.max(map_r.value), smap.data.shape[0] // 2) * u.R_sun
 
-    # Make sure bins are in the map.
+    # Ensure radial_bin_edges are within the bounds of the map_r values
     if radial_bin_edges[1, -1] > np.max(map_r):
-        radial_bin_edges = equally_spaced_bins(
-            inner_value=radial_bin_edges[0, 0],
-            outer_value=np.max(map_r),
-            nbins=radial_bin_edges.shape[1],
+        radial_bin_edges = (
+            equally_spaced_bins(
+                inner_value=radial_bin_edges[0, 0].to(u.R_sun).value,
+                outer_value=np.max(map_r.to(u.R_sun)).value,
+                nbins=radial_bin_edges.shape[1],
+            )
+            * u.R_sun
         )
-
     # Allocate storage for the filtered data
     data = np.zeros_like(smap.data)
-    meta = smap.meta
 
+    # Enable progress bar for large numbers of bins
     if radial_bin_edges.shape[1] > 2000:
         progress = True
 
-    # Calculate the filter values for each radial bin.
+    # Create figure and axis for animation if enabled
+    if animate:
+        fig, ax = plt.subplots(1, 1)
+        im = ax.imshow(data, origin="lower", cmap="gray", vmin=0, vmax=1)
+        plt.show(block=False)
+
+    # Select the ranking method
+    ranking_func = _select_rank_method(method)
+
+    # Loop over each radial bin to apply the filter
     for i in tqdm(range(radial_bin_edges.shape[1]), desc="RHEF: ", disable=not progress):
-        # Identify the appropriate radial slice
-        here = np.logical_and(map_r >= radial_bin_edges[0, i], map_r < radial_bin_edges[1, i])
+        # Identify pixels within the current radial bin
+        here = np.logical_and(
+            map_r >= radial_bin_edges[0, i].to(u.R_sun), map_r < radial_bin_edges[1, i].to(u.R_sun)
+        )
         if application_radius is not None and application_radius > 0:
             here = np.logical_and(here, map_r >= application_radius)
-
-        # Perform the filtering operation
-        ranking_func = _select_rank_method(method)
+        # Apply ranking function
         data[here] = ranking_func(smap.data[here])
         if upsilon is not None:
             data[here] = apply_upsilon(data[here], upsilon)
-    new_map = sunpy.map.Map(data, meta, autoalign=True)
 
+        # If animation is enabled, update the image data in the figure
+        if animate:
+            im.set_data(data)
+            plt.pause(0.0001)  # Pause to refresh the plot
+    if animate:
+        plt.show(block=True)
+
+    # Final update to the map
+    new_map = Map(data, smap.meta, autoalign=True)
```

Comment:
This autoalign does something?

Reply:
Probably not, I just saw it in an example.